### PR TITLE
Fix progress event output

### PR DESCRIPTION
**Pull request by Crydsch**
Fix for Issue #35

### CODE REVIEW

**Comment by rosenbjerg** [comment]
I don't see how this fixes progress events? But that would be nice :)
Edit: Just read your issue, makes sense now

**Comment by Crydsch** [comment]
Hey :D
The problem was that _totalTime was not set in the convert function.

The CI checks failing seem to be related to OSX?

**Comment by rosenbjerg** [comment]
Sounds good!
Yes, the CI failure seems to be unrelated to your changes

File: FFMpegCore/FFMPEG/FFMpeg.cs [hunk 1]
```diff
@@ -436,14 +436,21 @@ public VideoInfo ReplaceAudio(VideoInfo source, FileInfo audio, FileInfo output,
 
         public VideoInfo Convert(ArgumentContainer arguments)
         {
-            var args = ArgumentBuilder.BuildArguments(arguments);
             var output = ((OutputArgument)arguments[typeof(OutputArgument)]).GetAsFileInfo();
+            var sources = ((InputArgument)arguments[typeof(InputArgument)]).GetAsVideoInfo();
+
+            // Sum duration of all sources
+            _totalTime = TimeSpan.Zero;
+            foreach (var source in sources)
```
Comment (rosenbjerg) [inline FFMpegCore/FFMPEG/FFMpeg.cs:444]: Perhaps use  `_totalTime = sources.Sum(s => s.Duration)`?

File: FFMpegCore/FFMPEG/FFMpeg.cs [hunk 2]
```diff
@@ -436,14 +436,21 @@ public VideoInfo ReplaceAudio(VideoInfo source, FileInfo audio, FileInfo output,
...
+                _totalTime += source.Duration;
 
             if (!RunProcess(arguments, output))
             {
                 throw new FFMpegException(FFMpegExceptionType.Operation, "Could not replace the video audio.");
             }
 
+            _totalTime = TimeSpan.MinValue;
+
```
Comment (rosenbjerg) [inline FFMpegCore/FFMPEG/FFMpeg.cs:453]: Why is `_totalTime` reset before return?

**Review by rosenbjerg — CHANGES_REQUESTED** [review]
Looks good. Just a question and a suggestion

File: FFMpegCore/FFMPEG/FFMpeg.cs (hunk 1 above)
Comment (Crydsch) [reply FFMpegCore/FFMPEG/FFMpeg.cs:444]: TimeSpans can not be implicitly converted to long, But we could do
`_totalTime = new TimeSpan(sources.Sum(s => s.Duration.Ticks));`

File: FFMpegCore/FFMPEG/FFMpeg.cs (hunk 2 above)
Comment (Crydsch) [reply FFMpegCore/FFMPEG/FFMpeg.cs:453]: Actually i was unsure as well and copied it from the other convert method.
Maybe just remove both?

File: FFMpegCore/FFMPEG/FFMpeg.cs (hunk 1 above)
Comment (rosenbjerg) [reply FFMpegCore/FFMPEG/FFMpeg.cs:444]: Ah okay, I see. Just keep it as it is then :)

File: FFMpegCore/FFMPEG/FFMpeg.cs (hunk 2 above)
Comment (rosenbjerg) [reply FFMpegCore/FFMPEG/FFMpeg.cs:453]: No, just keep it. I just asked to hear if there was a reason for it being there :)

**Comment by rosenbjerg** [comment]
Looks good, I'll merge it

**Pull request merged**